lib/ansible/plugins/lookup/sops.py

```diff
@@ -0,0 +1,171 @@
+# -*- coding: utf-8 -*-
+#
+#  Copyright 2018 Edoardo Tenani <e.tenani@arduino.cc> [@endorama]
+#
+# This file is part of Ansible.
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from ansible.errors import AnsibleError, AnsibleParserError, AnsibleLookupError
+from ansible.plugins.lookup import LookupBase
+from ansible.module_utils._text import to_text, to_native
+
+from ansible.utils.display import Display
+display = Display()
+
+from subprocess import Popen, PIPE
+
+DOCUMENTATION = """
+    lookup: sops
+    author: Edoardo Tenani <e.tenani@arduino.cc>
+    version_added: "2.8"
+    short_description: Read sops encrypted file contents
+    description:
+        - This lookup returns the contents from a file on the Ansible controller's file system.
+        - This lookup requires the sops executable to be available in the controller PATH.
+    options:
+        _terms:
+            description: path(s) of files to read
+            required: True
+        print:
+            description: print the decrypted content
+            type: bool
+            required: False
+            default: False
+    notes:
+        - this lookup does not understand 'globing' - use the fileglob lookup instead.
+"""
+
+EXAMPLES = """
+tasks:
+  - name : output secrets to screen (BAD IDEA)
+    debug:
+        msg: "Content: {{lookup('sops', item)}}"
+    with_items:
+        - sops-encrypted-file.enc.yaml
+
+  - name: add ssh private key
+    copy:
+        content: "{{lookup('sops', user + '-id_rsa')}}"
+        dest: /home/{{user}}/.ssh/id_rsa
+        owner: "{{user}}"
+        group: "{{user}}"
+        mode: 0600
+"""
+
+RETURN = """
+    _raw:
+        description: decrypted file content
+"""
+
+# From https://github.com/mozilla/sops/blob/master/cmd/sops/codes/codes.go
+# Should be manually updated
+sops_error_codes = {
```

Comment:
You want to move that to module_utils once at least one of this and #59641 is merged.

Reply:
I will surely need some guidance on that, as I already tried but couldn't make it work. But sure, happy to tackle that.

Reply:
Essentially, it works as follows:
 1. Create a file `lib/ansible/module_utils/sops.py` which starts with the following, and add the code which is used by both plugins (i.e. `sops_error_codes`, `SopsError`, `Sops` from this PR):
```
# Copyright (c), xxx <yyy@zzz.com>, 2018-2020
# Simplified BSD License (see licenses/simplified_bsd.txt or https://opensource.org/licenses/BSD-2-Clause)

from __future__ import absolute_import, division, print_function
```
2. Add `from ansible.module_utils.sops import Sops, SopsError` to the plugin (and remove the code you've copied).

Reply:
Would you prefer this change to be in another PR or it's fine to add it to this one?

Reply:
I think it will be faster to merge this PR (since the other has the paths issue), so let's make the change here and get this merged, and adjust the other PR according (once this is merged). Does that sound good for you?